Sources/OpenTelemetrySdk/Trace/TracerProviderSdk.swift

```diff
@@ -66,16 +65,61 @@ public class TracerProviderSdk: TracerProvider {
         }
     }
 
+    /// Returns the active Clock.
+    public func getActiveClock() -> Clock {
+        return sharedState.clock
+    }
+
+    /// Updates the active Clock.
+    public func updateActiveClock(_ newClock: Clock) {
+        sharedState.clock = newClock
+    }
+
+    /// Returns the active IdGenerator.
+    public func getActiveIdGenerator() -> IdGenerator {
+        return sharedState.idGenerator
+    }
+
+    /// Updates the active IdGenerator.
+    public func updateActiveIdGenerator(_ newGenerator: IdGenerator) {
+        sharedState.idGenerator = newGenerator
+    }
+
+    /// Returns the active Resource.
+    public func getActiveResource() -> Resource {
+        return sharedState.resource
+    }
+
+    /// Updates the active Resource.
+    public func updateActiveResource(_ newResource: Resource) {
+        sharedState.resource = newResource
+    }
+
     /// Returns the active SpanLimits.
     public func getActiveSpanLimits() -> SpanLimits {
-        sharedState.activeSpanLimits
+        return sharedState.activeSpanLimits
     }
 
     /// Updates the active SpanLimits.
     public func updateActiveSpanLimits(_ spanLimits: SpanLimits) {
         sharedState.setActiveSpanLimits(spanLimits)
     }
 
+    /// Returns the active Sampler.
+    public func getActiveSampler() -> Sampler {
+        return sharedState.sampler
+    }
+
+    /// Updates the active Sampler.
+    public func updateActiveSampler(_ newSampler: Sampler) {
+        sharedState.setSampler(newSampler)
+    }
+
+    /// Returns the active SpanProcessor.
+    public func getActiveSpanProcessor() -> SpanProcessor {
```

Comment:
what does "active" span processor means? There may be many configured, correct? And there is not corresponding `updateActiveSpanProcessor` either

Reply:
Yes you are true, the name is not well chosen and maybe the functionality is not needed. I added it for having access to the span processors configured. In the case of having several span processors it would return the MultiSpanProcessor that wraps them. Maybe I should return the array of span processors included in the multispanprocessor instead and change the name. What do you think?

Reply:
I have reworked those methods, now they are: 
    `getActiveSpanProcessors()` returns an array with all the configured Span Processors
    `addSpanProcessor(_ spanProcessor: SpanProcessor)` add a new Span processor to the already configured 
   `resetSpanProcessors()` it removes all configured SpanProcessors and replace them with a Noop one